src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4203,7 +4203,22 @@ LogicalResult ONNXReduceSumSquareOp::inferShapes(
 
 LogicalResult ONNXRoiAlignOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  int64_t height = output_height();
+  int64_t width  = output_width();
+
+  auto xTy = X().getType().cast<RankedTensorType>();
```

Comment:
The input tensor may not be ShapedType in this pass of shape inference. You should check the condition before cast. If a tensor is not a ShapedType, simply `return success()`.